### PR TITLE
Fix/add jsx loading

### LINKED ISSUE #2127: [feature]: Ability to import and transpile JSX files
**Is your feature request related to a problem? Please describe.**
JSX is a common file type used when creating React components, and I would like to continue to use it as a file extension as it makes it much easier to identify React related JS from regular, non React JS, when looking through file directories

**Describe the solution you'd like**
The ability to import JSX files, without having to specify the JSX extension, and have Babel transpile files with the `.jsx` extension

**Describe alternatives you've considered**
Refactoring our existing code to all have `.js` extension. As described above, this creates an issue with code discoverability, in that it's not immediately obvious if a file contains React related code or if it's simply a JS utility file

**Please let us know what packages this feature is in regards to:**
- [x] `pwa-buildpack`

### PR DESCRIPTION
**Pull request by luke-denton-aligent**
## Description

This update adds the ability for people to use `.jsx` files with PWA Studio.
### Verification Steps
1. Create a new component, with `.jsx` file type
2. Import new component into an existing component, _without_ specifying the extension in the import statement, e.g. `import MyComponent from './MyComponent';`, where `MyComponent.jsx` exists alongside the file that the import statement is being added to
3. Verify that webpack doesn't throw an error indicating it can't resolve the path

## Related Issue
Closes #2127.

### Tests

Tests have been updated and run, with no errors
image

### CODE REVIEW

**Comment by larsroettig** [comment]
@luke-denton-aligent  very nice idea

File: packages/pwa-buildpack/lib/WebpackTools/MagentoResolver.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = {
             modules: [options.paths.root, 'node_modules'],
             mainFiles: ['index'],
             mainFields: ['esnext', 'es2015', 'module', 'browser', 'main'],
-            extensions: ['.wasm', '.mjs', '.js', '.json', '.graphql']
+            extensions: ['.wasm', '.mjs', '.js', '.jsx', '.json', '.graphql']
```
Comment (revanth0212) [inline packages/pwa-buildpack/lib/WebpackTools/MagentoResolver.js]: Doing this part of https://github.com/magento/pwa-studio/blob/a34adc181b065e83f884f7afbf29a2dd859dd8b9/jest.config.js#L68

**Comment by luke-denton-aligent** [comment]
@jimbo Hah, no worries, no JSX contributions!

Interesting that it's actually been disallowed internally, what's the reasoning behind that?

**Comment by dpatil-magento** [comment]
There are prettier failures, needs to be fixed.

**Comment by revanth0212** [comment]
> There are prettier failures, needs to be fixed.

Interesting, unable to reproduce locally.

image

**Comment by luke-denton-aligent** [comment]
Thanks for pointing that out @dpatil-magento, I've fixed the failure and udpated the PR, should be ready to go now :+1:

**Comment by dpatil-magento** [comment]
Thanks @luke-denton-aligent  @revanth0212 . Verification steps pass.

**Pull request merged**